staging/src/k8s.io/client-go/util/certificate/certificate_manager.go

```diff
@@ -445,6 +445,12 @@ func (m *manager) RotateCerts() (bool, error) {
 func (m *manager) rotateCerts() (bool, error) {
 	m.logf("%s: Rotating certificates", m.name)
 
+	// remove UsageKeyEncipherment before generating new certs as new certs is elliptic based
+	if hasKeyUsage(m.usages, certificates.UsageKeyEncipherment) {
+		// remove the last one as key encipherment was appened in kubelet.go
+		m.usages = m.usages[:len(m.usages)-1]
```

Comment:
this shouldn't assume kubelet.go is the only caller... I don't really think we should do this manipulation inside certificate manager, since that could be used by a lot of different clients talking to a lot of different API server versions

another approach could be to let the usage be dynamic based on the key, and let the kubelet set getUsage functions that only include encipherment for RSA keys. An example of that based on this PR is at https://github.com/liggitt/kubernetes/commits/key-encipherment-v1.26